src/runtime/src/core/map.rs

```diff
@@ -136,7 +139,25 @@ pub fn make_module() -> ValueMap {
 
     result.add_fn("sort", |vm, args| match vm.get_args(args) {
         [Map(m)] => {
-            m.contents_mut().data.sort_keys();
```

Comment:
It just occurred to me (and apologies for this not coming back to me sooner), but only immutable values are allowed as keys in `ValueMaps`, so it's actually not possible to have a type with overloaded `<` as a map key.

So, there's no need for customizing the sort here. Custom sorting is still needed in the 'sort with key function' case though because the key function can return an overloaded type.

Reply:
Sorry, I don't fully understand you here. How this method should work? `m.contents_mut().data.sort_keys();` won't work, because data isn't `PartialOrd`/`Ord`.

Reply:
`Value` _was_ `PartialOrd`+`Ord` until this branch, when I [commented out](https://github.com/koto-lang/koto/compare/operator-overloading-2#diff-0e87a461d9f7549944e6308db4defb1c443b01e73e6701fe891fe1963743fc63R174) the implementation to see what would break...

[`PartialEq`+`Eq`+`Hash` implementations](https://github.com/koto-lang/koto/blob/040c255d2170ac44e6743f6b381c7fb639e492a2/src/runtime/src/value.rs#L127)  are now only needed to allow `Value`s to be used as keys in `ValueMap`s, and only [immutable values](https://github.com/koto-lang/koto/blob/040c255d2170ac44e6743f6b381c7fb639e492a2/src/runtime/src/value.rs#L367) (which can't be overloaded) can be used as map keys, so I think it wouldn't be so bad to keep the `PartialOrd`+`Ord` implementations to support sorting here. 

These trait implementations will need to be cleaned up to make it explicit that they're only present for hash map support, and that comparisons should in general be performed via a vm (perhaps a newtype for 'values that can be hash map keys' could be extracted). This cleanup doesn't need to happen on this branch though, I'm just thinking out loud.

Reply:
So... Here I just need to uncomment `PartialOrd`/`Ord` implementation and return the previous implementation of `map.sort`, right?

Reply:
Yes I think that's the way to go. There are some types missing from the `PartialOrd`/`Ord` implementation (the types listed in `value_is_immutable` are the ones that can be used as map keys) but I don't mind if you leave them out for now and I can clean up these trait impls in a follow up.

Reply:
I've checked other value types and there are semantics you know better. For example, comparison of booleans, or external data. So I've uncommented the `PartialOrd`/`Ord` and brought back the old implementation, but kept the rest on you.